### ISSUE TITLE
SC2034 false positive when passing array variable names as function parameters

### ISSUE THREAD

**Issue by Nightfirecat** [koalaman/shellcheck#842]
In shellcheck 0.4.5, SC2034 is raised when passing array variable names as function parameters. This is done so an array variable can be expanded within the called function, effectively letting the name (`$1` in the function) be double-expanded.

```bash
cat << EOF > test.sh
#!/bin/bash
function print_array {
  array=("\${!1}")
  echo "\${array[@]}"
}
test_array=( 1   2   3 )
print_array test_array[@]
EOF
bash test.sh
shellcheck test.sh
```

#### Here's the current output:

```
1 2 3

In test.sh line 6:
test_array=( 1   2   3 )
^-- SC2034: test_array appears unused. Verify it or export it.
```

#### Here's what I wanted or expected to see:

```
1 2 3
```

**Comment by pikrzysztof** [comment]
MWE

![screenshot from 2018-07-10 18-49-50](https://user-images.githubusercontent.com/3964215/42527975-119c0396-8472-11e8-8db7-3e746f5e9291.png)

**Comment by matthewpersico** [comment]
It is unused. You spelled it wrong.
